### PR TITLE
add create_dirs kwarg to JSONStorage

### PR DESCRIPTION
**Pull request by Ofekmeister**
Setting create_dirs=True in `__init__` will create the necessary parent directories to ensure db file will be created. This will further lessen what the user has to worry about.

### CODE REVIEW

**Comment by Ofekmeister** [comment]
@msiemens Python 3.2 is weird and and py.test is errorring.

**Comment by eugene-eeo** [comment]
2 solutions I can think of:

 - pin to a specific version of `pytest`. Earlier the tests were not breaking on Python 3.2. travis already has `pytest` installed, so we probably need to remove [this line](https://github.com/msiemens/tinydb/blob/master/.travis.yml#L13) to see if travis installs a working pytest for us on 3.2.
 - drop 3.2 support altogether. It is more restrictive than 2.7 in terms of syntax (I think `b""` and `u""` is disallowed) and is the reason why we have to force ourselves to write strange, lowest denominator code.

**Comment by Ofekmeister** [comment]
@eugene-eeo Thanks for the feedback!

Should the database handle intermediate directory creation? Perhaps not, but it would be a useful option. Consider the main use case for a local db: an app storing data in a designated place, usually cross-platform.

```python
# an_app/db.py

import os
from appdirs import AppDirs

DATA_DIR = AppDirs('app_name').user_data_dir
DB_FILE = os.path.join(DATA_DIR, 'db_name.db')

# You will see this idiom in every program using a local datastore
if not os.path.exists(DATA_DIR):
    os.makedirs(DATA_DIR)
```

File: tests/test_storages.py [hunk 1]
```diff
@@ -71,6 +73,29 @@ def test_json_readwrite(tmpdir):
     assert get('A short one') is None
 
 
+def test_create_dirs():
+    temp_dir = tempfile.gettempdir()
+    db_dir = ''
+    db_file = ''
+
+    while True:
```
Comment (msiemens) [inline tests/test_storages.py]: I don't feel good about this loop. Couldn't this make tests hang for some edge cases? I'd propose using pytest's [tmpdir](http://doc.pytest.org/en/latest/tmpdir.html) that's also used in other tests (e.g. [test_storages.py#L49](https://github.com/msiemens/tinydb/blob/873a82e0b0ed71e6f2dcd3332e26960e95622675/tests/test_storages.py#L49)).

**Comment by msiemens** [comment]
> Should it really be the job of the database module to handle intermediate mkdirs?

Well, the way it's implemented it's the job of the storage which that seams reasonable to me. I don't have a strong opinion for/against a change like this but still think it's a useful addition.

**Comment by msiemens** [comment]
re Python 3.2: We don't support Python 3.0/3.1 either so I think it's fine to drop 3.2 too considering its usage is less than 1% (see pytest-dev/pytest#1627 for source + similar reasoning)

File: tests/test_storages.py (hunk 1 above)
Comment (Ofekmeister) [inline tests/test_storages.py]: @msiemens It will never hang. The loop just generates a temporary directory with the assurance that it does not already exist.

Comment (msiemens) [inline tests/test_storages.py]: Using pytest's tmpdir would allow us to not have to retry directory creation. Anyways, it's not a big deal really.

**Comment by Ofekmeister** [comment]
Would you like me to drop 3.2 in this PR?

**Comment by msiemens** [comment]
I'll do that in a later commit. @eugene-eeo I suppose you're fine with dropping Python 3.2?

**Comment by eugene-eeo** [comment]
@msiemens definitely. Python 3.2 for me is something that is 'midway' between 2 and 3. It doesn't have all the nice features of 3 but is also significantly different from 2.6+.

File: tests/test_storages.py (hunk 1 above)
Comment (eugene-eeo) [inline tests/test_storages.py]: It probably will not hang. If you however used something like `getrandbytes` it may cause issues with valid path characters and such. Also you don't have to include the clean up though, I believe tmpdirs are [cleaned up automatically anyways](http://doc.pytest.org/en/latest/tmpdir.html#the-default-base-temporary-directory).

**Comment by msiemens** [comment]
@eugene-eeo I'm a little confused, how does creating directories lead to corrupting user data?

**Comment by eugene-eeo** [comment]
@msiemens sorry, I have phrased it badly. It wouldn't be corruption but more of storing (and reading) incorrect data as it masks errors from invalid paths.

**Comment by Ofekmeister** [comment]
@eugene-eeo Can you please give an example?

**Comment by eugene-eeo** [comment]
@Ofekmeister @msiemens now that I've thought about it, it's not really an issue and I've confused myself, sorry. As it stands I'm ok with creating intermediate directories, as long as it is an _explicit_ option.

**Comment by msiemens** [comment]
Merged! Thanks everyone! I'll try to get a new release out sometime soon (this week or the next one)
